MobileWallet/Screens/Profile/ProfileViewController.swift

```diff
@@ -0,0 +1,188 @@
+*/
+
+import UIKit
+
+class ProfileViewController: UIViewController {
+
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var emojiContainer: UIView!
+    @IBOutlet weak var emojiLabel: UILabel!
+    @IBOutlet weak var copyEmojiButton: TextButton!
+    @IBOutlet weak var separatorView: UIView!
+    @IBOutlet weak var middleLabel: UILabel!
+    @IBOutlet weak var qrContainer: UIView!
+    @IBOutlet weak var qrImageView: UIImageView!
+
+    // MARK: - Override functions
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        customizeViews()
+    }
+
+    // MARK: - Private functions
+    private func customizeTitleLabel() {
+        let shareYourEmojiString = NSLocalizedString("Share your Emoji ID to receive Tari", comment: "profile title label")
+        let attributedString = NSMutableAttributedString(string: shareYourEmojiString,
+                                                         attributes: [.font: Theme.shared.fonts.profileTitleLightLabel!,
+                                                                      .foregroundColor: Theme.shared.colors.profileTitleTextColor!,
+                                                                      .kern: -0.33])
+        attributedString.addAttribute(.font,
+                                      value: Theme.shared.fonts.profileTitleRegularLabel!,
+                                      range: NSRange(location: 11, length: 8))
+
+        titleLabel.attributedText = attributedString
+    }
+
+    private func customizeEmoji() {
+        if let pubKey = TariLib.shared.tariWallet?.publicKey.0 {
+            let (emojis, _) = pubKey.emojis
+
+            self.emojiLabel.textColor = Theme.shared.colors.creatingWalletEmojisSeparator
+            self.emojiLabel.text = String(emojis.enumerated().map { $0 > 0 && $0 % 4 == 0 ? ["|", $1] : [$1]}.joined())
+        }
+    }
+
+    private func customizeCopyMyEmojiButton() {
+        copyEmojiButton.setVariation(.secondary)
+        let titleButton = NSLocalizedString("Copy my emoji ID", comment: "Profile title button")
+        self.copyEmojiButton.setTitle(titleButton, for: .normal)
+    }
+
+    private func customizeSeparatorView() {
+        separatorView.backgroundColor = Theme.shared.colors.profileSeparatorView!
+    }
+
+    private func customizeMiddleLabel() {
+        let middleLabelText = NSLocalizedString("Transacting in person? Your sender can get your emoji ID by scanning this QR code in their Tari app", comment: "Profile middle label")
+
+        self.middleLabel.text = middleLabelText
+        self.middleLabel.font = Theme.shared.fonts.profileMiddleLabel
+        self.middleLabel.textColor = Theme.shared.colors.profileMiddleLabel!
+    }
+
+    private func genQRCode() {
+
+        guard let wallet = TariLib.shared.tariWallet else {
+            return
+        }
+
+        let (walletPublicKey, pubKeyError) = wallet.publicKey
+        if pubKeyError != nil {
```

Comment:
Use a guard because there's no point generating the rest of the QR code without a public key